### PR TITLE
Adding a shape metric as loss

### PR DESCRIPTION
**Pull request by AndySAnker**
Adding a new loss function based on the shape of two scattering functions instead of their Euclidian distance: https://github.com/kiranvad/Amplitude-Phase-Distance/tree/main

### CODE REVIEW

File: src/ScattBO/utils/ScattBO.py [hunk 1]
```diff
@@ -273,6 +273,24 @@ def calculate_loss(x_target, x_sim, Int_target, Int_sim, loss_type="rwp"):
...
+            optim_kwargs = {"optim": "DP", "grid_dim": 10}
+            t = (x_target - np.max(x_target)) / (np.max(x_target) - np.min(x_target))
+
+            # Calculate phase_dist and amplitude_dist using the dist function
+            phase_dist, amplitude_dist = dist(t, Int_target_double, Int_sim_interp_double, **optim_kwargs)
+
+            # Convert amplitude_dist to a torch tensor for loss
+            loss = torch.tensor(amplitude_dist, dtype=torch.float32)
+        except ModuleNotFoundError:
+            print("Follow the installation instructions at https://github.com/kiranvad/Amplitude-Phase-Distance/tree/main")
```
Comment (miguelgondu) [inline src/ScattBO/utils/ScattBO.py:293]: Hey Andy!

One small question: shouldn't we raise the issue here? Otherwise, there's no loss and the program continues.

I'd maybe change the print statement for a `raise ModuleNotFoundError("Follow the installation instructions at ...")`.
